pkg/lite/lite.go

```diff
@@ -415,11 +452,46 @@ func start(ctx context.Context, opts StartOpts) error {
 		HistoryReader:   hr,
 		LocalSigningKey: opts.SigningKey,
 		RequireKeys:     true,
+		ConnectOpts: connectv0.Opts{
+			GroupManager:            connectionManager,
+			ConnectManager:          connectionManager,
+			ConnectResponseNotifier: apiConnectProxy,
+			Signer:                  auth.NewJWTSessionTokenSigner(consts.DevServerConnectJwtSecret),
+			RequestAuther:           ds,
+			ConnectGatewayRetriever: ds,
+			EntitlementProvider:     ds,
+			ConditionalTracer:       conditionalTracer,
+		},
 	})
 	if err != nil {
 		return err
 	}
 
+	gatewayRequestReceiver, err := connectpubsub.NewConnector(ctx, connectpubsub.WithRedis(connectRcOpt, connectPubSubLogger, conditionalTracer, false))
+	if err != nil {
+		return fmt.Errorf("failed to create connect pubsub connector: %w", err)
+	}
+
+	connGateway := connect.NewConnectGatewayService(
+		connect.WithConnectionStateManager(connectionManager),
+		connect.WithRequestReceiver(gatewayRequestReceiver),
+		connect.WithGatewayAuthHandler(auth.NewJWTAuthHandler(consts.DevServerConnectJwtSecret)),
+		connect.WithDev(),
+		connect.WithGatewayPublicPort(opts.ConnectGatewayPort),
+		connect.WithApiBaseUrl(fmt.Sprintf("http://127.0.0.1:%d", opts.Config.EventAPI.Port)),
```

Comment:
Two notes:
* Should this hostname come from the config instead of hard coded? ex. `opts.Config.CoreAPI.Addr` Is the hostname. 
* Should the port be `opts.Config.CoreAPI.Port` instead of `EventAPI`? I know they're the same currently, but they could change in the future.

Looking at `start.go`, we also would need to set the CoreAPI port/host in additional to the eventapi port/host for this to work I think. Making host driven from config/ops could fix a couple weird things for self-hosting on different hostnames within a network.

Reply:
Can we also pass this API or hostname into the `RetrieveGateway` function in `service.go`? Right now it's hard coded to `127.0.0.1`, but pushing the `host` flag/config option would allow the dev server and start to work without necessarily needing `rewriteGatewayEndpoint` at all.